### ISSUE TITLE
3.178.0 may break @aws-sdk/s3-request-presigner getSignedUrl

### ISSUE THREAD

**Issue by bmccann36** [aws/aws-sdk-js-v3#3983]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

3 hours ago there was a release to version 3.178.0. We are using `npm i` not `ci` in our pipeline so a newer version was installed. We did not change anything in our code but after the new build we started seeing this when our code tried to execute `import {getSignedUrl} from '@aws-sdk/s3-request-presigner'`
```
2022-09-23T21:25:54.847Z	undefined	ERROR	Uncaught Exception 	{
    "errorType": "Runtime.ImportModuleError",
    "errorMessage": "Error: Cannot find module '@aws-sdk/middleware-endpoint'\nRequire stack:\n- /opt/nodejs/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/getSignedUrl.js\n- /opt/nodejs/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/index.js\n- /var/task/api/services/businessLogic/asset-logic.js\n- /var/task/api/services/businessLogic/inbound-ticket/inbound-ticket-logic.js\n- /var/task/handlers/inbound-ticket-handler.js\n- /var/runtime/UserFunction.js\n- /var/runtime/Runtime.js\n- /var/runtime/index.js",
    "stack": [
        "Runtime.ImportModuleError: Error: Cannot find module '@aws-sdk/middleware-endpoint'",
        "Require stack:",
        "- /opt/nodejs/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/getSignedUrl.js",
        "- /opt/nodejs/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/index.js",
```
If we change our package.json from
`"@aws-sdk/s3-request-presigner": "^3.178.0"` => `"@aws-sdk/s3-request-presigner": "3.67.0"` 
it fixes the issue

### SDK version number

@aws-sdk/s3-request-presigner@3.178.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

node 14 on AWS Lambda

### Reproduction Steps

- make a typescript node project
- install dependencies
- execute a file with these contents in node
```typescript
import {getSignedUrl} from '@aws-sdk/s3-request-presigner';
import {GetObjectCommand, S3Client,} from '@aws-sdk/client-s3';
const client = new S3Client({});
const command = new GetObjectCommand({Key: '123', Bucket: 'abc'});
getSignedUrl(client, command).then(console.log)
```

### Observed Behavior

```
➜  scratch node --version
v16.17.0
➜  scratch node scratch.js
node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module '@aws-sdk/middleware-endpoint'
Require stack:
- /Users/brianmccann/Desktop/scratch/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/getSignedUrl.js
- /Users/brianmccann/Desktop/scratch/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/index.js
- /Users/brianmccann/Desktop/scratch/scratch.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/brianmccann/Desktop/scratch/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/getSignedUrl.js:4:31)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/brianmccann/Desktop/scratch/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/getSignedUrl.js',
    '/Users/brianmccann/Desktop/scratch/node_modules/@aws-sdk/s3-request-presigner/dist-cjs/index.js',
    '/Users/brianmccann/Desktop/scratch/scratch.js'
  ]
}

```

### Expected Behavior

I am able to call `getSignedUrl`

**Comment by bmccann36** [comment]
looks like this is a dupe of https://github.com/aws/aws-sdk-js-v3/issues/3981

**Comment by ahmed-sharief5** [comment]
Even i am facing the same issue. Could anyone resolve this

**Comment by mdesousa** [comment]
as a workaround while the issue is fixed, we've added `@aws-sdk/middleware-endpoint` to our package.json and that solved the problem

**Comment by ianwremmel** [comment]
> as a workaround while the issue is fixed, we've added `@aws-sdk/middleware-endpoint` to our package.json and that solved the problem

I tried this and discovered other packages also seem to be missing from dep tree.

**Comment by appleinautumn** [comment]
> [mdesousa](/mdesousa)
mdesousa's workaround works.

**Comment by appleinautumn** [comment]
> as a workaround while the issue is fixed, we've added `@aws-sdk/middleware-endpoint` to our package.json and that solved the problem

mdesousa's workaround works.

**Comment by thangsotanext** [comment]
This is exactly what I got today.
Thank you

**Comment by thangchinh** [comment]
This is exactly what I got today. Change to older version fix this problem.

**Comment by ahmed-sharief5** [comment]
Even after installing the package separately it shows Cannot find module. Could the owner fix this issue please. Its causing our application to down from 3 days

**Comment by aitchkhan** [comment]
@ahmed-sharief5 try using an older version.

**Comment by earthbrands** [comment]
I have Nodejs code running on aws fargate container. What I did just add one logic, tested code and pushed code to github. And AWS started pipeline process. When deployment failed I checked cloud watch to know why its failed. So I found that my Nodejs code requrire one package to run @aws-sdk/middleware-endpoint. When I tested my nodejs code in my local system I didn't received any such error.
After reading this error I installed package @aws-sdk/middleware-endpoint and my server is running well. I am willing to know why @aws-sdk/middleware-endpoint is required suddenly. Although previously my all apis are working fine in same aws server, But today I saw requirement of @aws-sdk/middleware-endpoint.

Anyone can help me what is need of @aws-sdk/middleware-endpoint?

**Comment by imkumawat** [comment]
I have Nodejs code running on aws fargate container. What I did just add one logic, tested code and pushed code to github. And AWS started pipeline process. When deployment failed I checked cloud watch to know why its failed. So I found that my Nodejs code requrire one package to run @aws-sdk/middleware-endpoint. When I tested my nodejs code in my local system I didn't received any such error.
After reading this error I installed package @aws-sdk/middleware-endpoint and my server is running well. I am willing to know why @aws-sdk/middleware-endpoint is required suddenly. Although previously my all apis are working fine in same aws server, But today I saw requirement of @aws-sdk/middleware-endpoint.

Anyone can help me what is need of @aws-sdk/middleware-endpoint?
![1](https://user-images.githubusercontent.com/86528693/192346698-04eeedfd-63a1-4bd9-abbc-83fabfd3ca72.PNG)
